### PR TITLE
feature:[JIRATASK-000] Added Error Boundary

### PR DESCRIPTION
**Pull request by dasarisamanth**
Have added an ErrorBoundary Component using react-error-boundary package.
Wrote a basic assertion for the same.
Also have added a basic Fallback UI/NotFound Page will improvise the UI once we have design and necessary assets.

### CODE REVIEW

**Comment by EvgeniyKvasyuk** [comment]
looks well, but need to resolve conflicts and use ErrorBoundary in src/index.ts in this PR

File: src/modules/NotFound/NotFound.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+import { Typography } from '@mui/material';
```
Comment (VladSolodjankin) [inline src/modules/NotFound/NotFound.tsx:3]: I think it is better to move it to common components and write some abstraction over it
Also don't forget to and storybook

**Comment by dasarisamanth** [comment]
@EvgeniyKvasyuk Have added ErrorBoundary to src/index.tsx

File: src/modules/NotFound/NotFound.tsx (hunk 1 above)
Comment (dasarisamanth) [reply src/modules/NotFound/NotFound.tsx]: @VladSolodjankin Moved it to common please take a look and I'll add the stories soon in the next PR

Comment (VladSolodjankin) [reply src/modules/NotFound/NotFound.tsx]: sorry looks like I confused you. We don't use  @mui/material directly in project, We write wrappers over it. You can look on Icon as an example.

Comment (dasarisamanth) [reply src/modules/NotFound/NotFound.tsx]: @VladSolodjankin Updated the PR

File: src/components/Typography/Typography.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import TypographyMaterial, { TypographyProps as TypographyPropsMaterial } from '@mui/material/Typography';
+import classNames from 'classnames/bind';
+
+import styles from './Typography.module.css';
+
+const cn = classNames.bind(styles);
+
+export type TypographyProps = {
+  className?: string;
+} & Pick<TypographyPropsMaterial, 'children' | 'className' | 'variant' | 'align'>;
```
Comment (VladSolodjankin) [inline src/components/Typography/Typography.tsx:12]: you can delete className here because you already declareted it at the top

**Review by VladSolodjankin — APPROVED** [review]

File: src/components/Typography/Typography.tsx (hunk 2 above)
Comment (dasarisamanth) [reply src/components/Typography/Typography.tsx]: Done

**Pull request merged**